### PR TITLE
Drop references to the old docs on the Welcome page.

### LINKED ISSUE #344: Update "Welcome" introduction as the old documents no longer exist
### What did you do?
I visited the first page (https://developers.stellar.org/docs/) and noticed the Welcome introduction is outdated.

If possible, provide a recipe for reproducing the error.
A complete working example or script is great.
A link on an online code editor is best. The following online editors are great.
 - JS: https://runkit.com or https://jsfiddle.net
 - Go: https://play.golang.org
 - Others: https://repl.it
-->

### What did you expect to see?
An update to the language now that the old documents no longer exist.

### What did you see instead?
The old documents are deprecated but are still being referred to in the text as the existing documents, and the existing documents are being referred to as the new documents:  

"Right now, you’re looking at the **new** Stellar docs, which are a work in progress. These **new** docs will live alongside the **existing** Stellar docs until we complete the content, do some polish passes, root out and squash any bugs in the interface, and get feedback from the SDF and the Stellar community at large.

The goal is to keep everything good from the **existing** docs but to streamline the navigation, improve the looks, and update the content to better reflect where things stand now. These **new** docs are also more modular — they’re organized around different learning paths — and the plan is to add new paths as network use cases evolve.

As you read through these **new** docs, you will likely find bugs, errors, typos, or omissions. When you do, please file PRs or issues in this repo or make a report using this typeform survey."

### PR DESCRIPTION
**Pull request by Shaptic**
Closes #344.

### CODE REVIEW

File: content/docs/index.mdx [hunk 1]
```diff
@@ -5,13 +5,11 @@ order: -100
...
+In addition to the docs here, you'll also see a link at the bottom of the left nav to the API Reference. You should check those out, too: that's a resource most developers consult as they build on Stellar.
 
-As you read through these new docs, you will likely find bugs, errors, typos, or omissions. When you do, please file PRs or issues in this repo or make a report using [this typeform survey](https://stellarform.typeform.com/to/suEIZI).
-
-In addition to the docs here, which are guides to building on Stellar, you'll also see a link at the bottom of the left nav to the API Reference. You should check those out, too: that's a resource most developers consult as they build on Stellar.
+As you read through these docs, you may find bugs, errors, typos, or omissions. When you do, please file PRs or issues in [this repo](https://github.com/stellar/new-docs/issues/new) or make a report using [this typeform survey](https://stellarform.typeform.com/to/suEIZI).
```
Comment (rice2000) [inline content/docs/index.mdx:12]:
```suggestion
As you read through these docs, you may find bugs, errors, typos, or omissions. When you do, please file PRs or issues in [this repo](https://github.com/stellar/new-docs/issues/new).  Like the Stellar codebase, the docs are open source, and we welcome and appreciate your contributions!
```

Comment (rice2000) [reply content/docs/index.mdx:12]: This looks good.  Suggestion added inline.  Basic idea:
- Let's kill the typeform at this point so the repo is the source of truth
- And say something friendly to encourage contributions.

**Review by rice2000 — CHANGES_REQUESTED** [review]

File: content/docs/index.mdx (hunk 1 above)
Comment (Shaptic) [reply content/docs/index.mdx]: Good call on this! I like `so` over `and` as a conjunction to not clash with the subsequent `and`; I did this in 1035528.

Comment (rice2000) [reply content/docs/index.mdx]: Agreed!  Good call.

**Review by rice2000 — APPROVED** [review]
Perfecto.

**Pull request merged**